pkg/kubelet/kubelet_getters_test.go

```diff
@@ -91,3 +96,111 @@ func TestKubeletDirs(t *testing.T) {
 	exp = filepath.Join(root, "pod-resources")
 	assert.Equal(t, exp, got)
 }
+
+func TestGetPod(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet, expectedPods, expectedRunningPods := newKubeletAndPods(testKubelet)
+
+	testCases := []struct {
+		name       string
+		assertFunc func(t *testing.T)
+	}{
+		{
+			name: "GetPods",
+			assertFunc: func(t *testing.T) {
+				got := kubelet.GetPods()
+				assert.Equal(t, len(expectedPods), len(got))
+			},
+		},
+		{
+			name: "GetRunningPods",
+			assertFunc: func(t *testing.T) {
+				got, err := kubelet.GetRunningPods()
+				if err != nil {
+					t.Fatalf("unexpected error: %v", err)
+				}
+				assert.Equal(t, len(expectedRunningPods), len(got))
+			},
+		},
+		{
+			name: "GetPodByFullName",
+			assertFunc: func(t *testing.T) {
+				got, ok := kubelet.GetPodByFullName("pod0_")
+				assert.Equal(t, true, ok)
+				assert.Equal(t, expectedPods[0], got)
+			},
+		},
+		{
+			name: "GetPodByName",
+			assertFunc: func(t *testing.T) {
+				got, ok := kubelet.GetPodByName("", "pod1")
+				assert.Equal(t, true, ok)
+				assert.Equal(t, expectedPods[1], got)
+			},
+		},
+		{
+			name: "GetPodByCgroupfs",
+			assertFunc: func(t *testing.T) {
+				got, ok := kubelet.GetPodByCgroupfs("10001")
+				assert.Equal(t, false, ok)
+				assert.Equal(t, (*v1.Pod)(nil), got)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.assertFunc(t)
+		})
+	}
+}
+
+func TestGetHost(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+
+	got := kubelet.GetHostname()
+	exp := "127.0.0.1"
+	assert.Equal(t, exp, got, "GetHostname")
+
+	gotIPs, err := kubelet.GetHostIPs()
+	assert.NoError(t, err, "GetHostIPs error")
+	expIPs := []net.IP{
+		net.ParseIP("127.0.0.1"),
+	}
+	assert.Equal(t, expIPs, gotIPs, "GetHostIPs")
+
+	kubelet.nodeName = "127.0.0.2"
+	_, err = kubelet.GetHostIPs()
+	expErr := fmt.Errorf("Node with name: 127.0.0.2 does not exist")
+	assert.ErrorAs(t, err, &expErr, "GetHostIPs")
+}
+
+func newKubeletAndPods(testKubelet *TestKubelet) (*Kubelet, []*v1.Pod, []*containertest.FakePod) {
+	kubelet := testKubelet.kubelet
+
+	expectedPods := newTestPods(2)
```

Comment:
This doesn't create static pods, but we have logic that only triggers on static pod. It would be good to include a test case with static pods.

Reply:
Well, the two lines after this line construct pod[0] as a static pod, in order to make `GetPods` cover wider.
In addition, `but we have logic that only triggers on static pod` 
refers to the isStaticPod or isMirrorPod in pod_update.go?

 I did not find `getter.go` has logic similar to GetStaticPods, can you specify which function needs to be tested? :-)

Reply:
I missed that, this is fine.